src/actions/PetitionsActions.js

```diff
@@ -29,19 +29,26 @@ export function fetchPetitions ({ location, params }, group) {
       .then(response => dispatch(receivePetitions(
         response,
         queryParams,
-        queryString,
-        group
+        queryString
       )));
   };
 }
 
 export function fetchGroupedPetitions (props = {}, groups = []) {
-  const sortQuery = getHomePagePetitionsQuery();
-  const location = Object.assign({}, props.location, sortQuery);
-
   return (dispatch) => {
-    const groupDispatches = groups.map(group =>
-      dispatch(fetchPetitions({ location }, group)));
+    dispatch(requestPetitions());
```

Comment:
Yep, you're right. Each group needs its own `isLoading` property. And  the `<Loading />` component needs to be wrapped around the homepage petitions...